feat(releasedefn): generate release defintion based on a yaml configuration

azlam-abdulsalam opened the pull request:
Deprecate the existing flags and generate release defn based on a yaml configuration file. This
allows for much greater flexibility and also allows the user to exclude artifacts with ease.  Please
note excludeArtifactsWithTags is not implemented yet

BREAKING CHANGE: existing flags on the commands are deprecated








#### Checklist
All items have to be completed before a PR is merged

- [x] Adhere to [Contribution Guidelines](https://docs.dxatscale.io/about-us/contributing-to-dx-scale)
- [x] Updates to Decision Records considered?
- [ ] Updates to documentation at [DX@Scale Guide](https://github.com/dxatscale/dxatscale-guide) considered?
- [x] Tested changes?
- [x] Unit Tests new and existing passing locally?

Commit:
feat(releasedefn): generate release defintion based on a yaml configuration

Deprecate the existing flags and generate release defn based on a yaml configuration file. This
allows for much greater flexibility and also allows the user to exclude artifacts with ease.  Please
note excludeArtifactsWithTags is not implemented yet

BREAKING CHANGE: existing flags on the commands are deprecated

Commit:
feat(releasedefn): generate release defintion based on a yaml configuration

Deprecate the existing flags and generate release defn based on a yaml configuration file. This
allows for much greater flexibility and also allows the user to exclude artifacts with ease.  Please
note excludeArtifactsWithTags is not implemented yet

BREAKING CHANGE: existing flags on the commands are deprecated

Commit:
fix(imports): remove unused import

Commit:
fix(imports): remove unused import

Commit:
fix(releasedefn): handle empty arrays when the value is not set

Commit:
fix(releasedefn): handle empty arrays when the value is not set

Schuchie commented on packages/sfpowerscripts-cli/src/impl/release/ReleaseDefinitionGenerator.ts:
```diff
@@ -151,6 +176,59 @@ export default class ReleaseDefinitionGenerator {
+            releaseName = name.replace(/[/\\?%*:|"<>]/g, '-').concat(`-`, buildNumber.toString());
+            return releaseName;
+        } else {
+            return this.releaseDefinitionGenratorConfigSchema.releaseName;
```
Typo: Generator instead of Genrator

azlam-abdulsalam replied:
Thanks fixed

Commit:
wip(releasedefn): add ability to add artifacts

Commit:
wip(releasedefn): add ability to add artifacts

Commit:
fix(releasdefn): fix typo with schema files

Commit:
fix(releasdefn): fix typo with schema files

zhebinliu committed:
feat(build): add an option to specify specific config file paths (#1093)

This feature adds ability to use multiple scratch org definition files, where each package can be specified with a particular Definition

Commit:
fix(relasedefn): add independent usage of flags with inclusion filter

Commit:
fix(relasedefn): add independent usage of flags with inclusion filter

azlam-abdulsalam committed:
Merge branch 'develop' into feat/release-defn-generator-schema

Commit:
feat(releasedefn): add ability to override name

Commit:
Merge branch 'feat/release-defn-generator-schema' of https://github.com/Accenture/sfpowerscripts into feat/release-defn-generator-schema

Commit:
fix(releasedefn): rename the schema to to emphasize 'only'
